src/providers/tokens.provider.ts

```diff
@@ -249,10 +249,13 @@ export const tokensProvider = (
     addresses = [...new Set(addresses)];
 
     const existingAddresses = Object.keys(tokens.value);
+    const existingAddressesMap = Object.fromEntries(
+      existingAddresses.map((address: string) => [getAddress(address), true])
+    );
 
     // Only inject tokens that aren't already in tokens
     const injectable = addresses.filter(
-      address => !includesAddress(existingAddresses, address)
+      address => existingAddressesMap[address] !== true
```

Comment:
prob just `!existingAddressesMap[address]` would be better?
I get a little bit of brain lag when see`!== true` in filter callback 😁